In [2]:
import numpy as np 
import pandas as pd
import seaborn as sns

In [3]:
df = pd.read_csv(r'C:\\Users\\Master Ridwan\\Desktop\\my data\\FoodBalanceSheets_E_Africa_NOFLAG.csv', encoding = 'latin-1' )

In [25]:
df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [8]:
df.shape

(60943, 12)

### which year has the least correlation with 'Element Code'?

In [5]:
df_corr = df[['Element Code', 'Y2014','Y2015', 'Y2016', 'Y2017', 'Y2018']]

In [6]:
df_corr

Element Code     Y2014     Y2015     Y2016     Y2017     Y2018
0               511  38924.00  39728.00  40551.00  41389.00  42228.00
1              5301      0.00      0.00      0.00      0.00      0.00
2               664   3377.00   3379.00   3372.00   3341.00   3322.00
3               674     94.90     94.35     94.72     92.82     91.83
4               684     80.06     79.36     77.40     80.19     77.28
...             ...       ...       ...       ...       ...       ...
60938          5142     42.00     46.00     33.00     19.00     16.00
60939           645      3.06      3.33      2.35      1.33      1.08
60940           664      3.00      4.00      3.00      1.00      1.00
60941           674      0.10      0.11      0.08      0.04      0.04
60942           684      0.04      0.05      0.03      0.02      0.01

[60943 rows x 6 columns]

In [7]:
df_corr.corr()

Element Code     Y2014     Y2015     Y2016     Y2017     Y2018
Element Code      1.000000  0.024457  0.023889  0.023444  0.024254  0.024279
Y2014             0.024457  1.000000  0.994647  0.996081  0.995230  0.994872
Y2015             0.023889  0.994647  1.000000  0.995739  0.988048  0.988208
Y2016             0.023444  0.996081  0.995739  1.000000  0.992785  0.992757
Y2017             0.024254  0.995230  0.988048  0.992785  1.000000  0.998103
Y2018             0.024279  0.994872  0.988208  0.992757  0.998103  1.000000

The year with the least correlation with 'Element Code' is 2016

## which area has the lowest sum in 2017(perform a groupby operation)?

In [9]:
df['Area'].unique()

array(['Algeria', 'Angola', 'Benin', 'Botswana', 'Burkina Faso',
       'Cabo Verde', 'Cameroon', 'Central African Republic', 'Chad',
       'Comoros', 'Congo', "Côte d'Ivoire", 'Djibouti', 'Egypt',
       'Eswatini', 'Ethiopia', 'Ethiopia PDR', 'Gabon', 'Gambia', 'Ghana',
       'Guinea', 'Guinea-Bissau', 'Kenya', 'Lesotho', 'Liberia',
       'Madagascar', 'Malawi', 'Mali', 'Mauritania', 'Mauritius',
       'Morocco', 'Mozambique', 'Namibia', 'Niger', 'Nigeria', 'Rwanda',
       'Sao Tome and Principe', 'Senegal', 'Seychelles', 'Sierra Leone',
       'South Africa', 'Sudan', 'Sudan (former)', 'Togo', 'Tunisia',
       'Uganda', 'United Republic of Tanzania', 'Zambia', 'Zimbabwe'],
      dtype=object)

In [11]:
gtp = df.groupby(['Area', 'Y2017'])

In [15]:
gtp.first()

Area Code  Item Code                          Item  \
Area     Y2017                                                          
Algeria  -81.0             4       2919       Fruits - Excluding Wine   
         -47.0             4       2907                 Starchy Roots   
         -11.0             4       2563  Olives (including preserved)   
         -6.0              4       2680                   Infant food   
         -5.0              4       2918                    Vegetables   
...                      ...        ...                           ...   
Zimbabwe  1891.0         181       2901                   Grand Total   
          2209.0         181       2905      Cereals - Excluding Beer   
          3101.0         181       2908                   Sugar Crops   
          3452.0         181       2908                   Sugar Crops   
          14237.0        181       2501                    Population   

                   Element Code                        Element  \
Area     Y2017                                                   
Algeria  -81.0             5072                Stock Variation   
         -47.0             5072                Stock Variation   
         -11.0             5072                Stock Variation   
         -6.0              5072                Stock Variation   
         -5.0              5170                      Residuals   
...                         ...                            ...   
Zimbabwe  1891.0            664  Food supply (kcal/capita/day)   
          2209.0           5301       Domestic supply quantity   
          3101.0           5511                     Production   
          3452.0           5131                     Processing   
          14237.0           511  Total Population - Both sexes   

                              Unit    Y2014    Y2015    Y2016    Y2018  
Area     Y2017                                                          
Algeria  -81.0         1000 tonnes     28.0     -3.0    -63.0      0.0  
         -47.0         1000 tonnes    156.0     35.0     61.0    -90.0  
         -11.0         1000 tonnes    -29.0    -11.0     23.0     34.0  
         -6.0          1000 tonnes      1.0      4.0      1.0      0.0  
         -5.0          1000 tonnes     -4.0     -3.0     -5.0     -2.0  
...                            ...      ...      ...      ...      ...  
Zimbabwe  1891.0   kcal/capita/day   1986.0   1912.0   1885.0   1908.0  
          2209.0       1000 tonnes   2013.0   1851.0   1940.0   2071.0  
          3101.0       1000 tonnes   3856.0   3348.0   3483.0   3560.0  
          3452.0       1000 tonnes   3785.0   3552.0   3728.0   3669.0  
          14237.0     1000 persons  13587.0  13815.0  14030.0  14439.0  

[14773 rows x 10 columns]

Area with the lowest sum in 2017 is ALgeria

In [ ]:
#df.groupby['']

## which is the total number of missing value data in 2014?

In [21]:
df['Y2014'].isnull().sum()

1589

 The total number of missing value data in 2014 is 1589

## what is the percentage of missing data in 2014?

In [23]:
df['Y2014'].isnull().mean()*100

2.6073544131401474

The percentage of missing data in 2014 is 2.6073544131401474

## what is the number of unique areas?

In [24]:
df['Area'].nunique()

49

The number of unique area is 49

## what is the total number of processing in 2017(perform a groupby on element, and perform sum operation on 2017) ?

In [68]:
df.groupby('Element')['Y2017'].sum()

Element
Domestic supply quantity                  2088198.10
Export Quantity                            182338.80
Fat supply quantity (g/capita/day)          10253.84
Feed                                       223705.68
Food                                      1258888.28
Food supply (kcal/capita/day)              454681.00
Food supply quantity (kg/capita/yr)         48690.04
Import Quantity                            294559.09
Losses                                     160614.00
Other uses (non-food)                       91645.97
Processing                                 292836.00
Production                                2030056.89
Protein supply quantity (g/capita/day)      11842.45
Residuals                                   35500.00
Seed                                        24870.14
Stock Variation                             54316.91
Total Population - Both sexes             1112641.00
Tourist consumption                            91.00
Name: Y2017, dtype: float64

total number of processing in 2017 292836.00

## what is the total amount of wine produced in 2015 and 2016?

In [30]:
df['Y2015'].sum() + df['Y2016'].sum() 

16144812.879999999

total amount of wine produced in 2015 and 2016 is 16144812.879999999

## what area had he highest sum in 2017?

In [62]:
df.groupby('Area')["Y2017"].sum()

Area
Algeria                         325644.27
Angola                          229159.57
Benin                           124771.22
Botswana                         22101.30
Burkina Faso                    101855.07
Cabo Verde                       14650.74
Cameroon                        232030.43
Central African Republic         29937.00
Chad                             71594.68
Comoros                             59.84
Congo                            41181.68
Côte d'Ivoire                   224599.01
Djibouti                         22729.91
Egypt                           866379.92
Eswatini                         54343.33
Ethiopia                        448683.76
Ethiopia PDR                         0.00
Gabon                            27979.64
Gambia                           23154.18
Ghana                           337599.06
Guinea                           98138.87
Guinea-Bissau                    19102.77
Kenya                           264660.66
Lesotho                      

the area with the highest sum in 2017 is Nigeria